portal/integrated_windows_authentication/src/main/java/com/esri/samples/integrated_windows_authentication/IntegratedWindowsAuthenticationController.java

```diff
@@ -0,0 +1,202 @@
+          String selectedItemId = resultsListView.getSelectionModel().getSelectedItem().getKey();
+
+          // use the item ID to create a PortalItem from the portal
+          PortalItem portalItem = new PortalItem(iwaSecuredPortal, selectedItemId);
+
+          if (portalItem != null) {
+            // create a Map using the web map (portal item)
+            ArcGISMap webMap = new ArcGISMap(portalItem);
+            // set the map to the map view
+            mapView.setMap(webMap);
+          }
+
+          loadWebMapTextView.setText("Loaded web map from item " + selectedItemId);
+        }
+      });
+
+      // make the list view show a preview of the portal items' map area
+      resultsListView.setCellFactory(c -> new PortalItemInfoListCell());
+
+      // hide the progress indicator when the map is finished drawing
+      mapView.addDrawStatusChangedListener(drawStatusChangedEvent -> {
+        if (drawStatusChangedEvent.getDrawStatus() == DrawStatus.COMPLETED) {
+          progressIndicator.setVisible(false);
+        }
+      });
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Handles searching the provided IWA secured portal.
+   */
+  @FXML
+  private void handleSearchPortalPress() {
+    // check for a url in the URL field
+    if (!portalUrlTextField.getText().isEmpty()) {
+      // keep hold of the portal we are searching and set a variable indicating that this is a secure portal, to allow retrieving portal items later
+      iwaSecuredPortal = new Portal(portalUrlTextField.getText(), true);
+
+      // clear any existing items in the list view
+      resultsListView.getItems().clear();
+
+      // clear the information about the previously loaded map
+      loadWebMapTextView.setText("");
+
+      // show portal load state
+      progressIndicator.setVisible(true);
+      loadStateTextView.setText("Searching for web map items on the portal at " + iwaSecuredPortal.getUri());
+
+      // load the portal items
+      iwaSecuredPortal.loadAsync();
+      iwaSecuredPortal.addDoneLoadingListener(() -> {
+        if (iwaSecuredPortal.getLoadStatus() == LoadStatus.LOADED) {
+          try {
+            // update load state in UI with the portal URI
+            loadStateTextView.setText("Connected to the portal on " + new URI(iwaSecuredPortal.getUri()).getHost());
+          } catch (URISyntaxException e) {
+            new Alert(Alert.AlertType.ERROR, "Error getting URI from portal: " + e.getMessage()).show();
+          }
+
+          // report the user name used for this connection
+          if (iwaSecuredPortal.getUser() != null) {
+            loadStateTextView.setText("Connected as: " + iwaSecuredPortal.getUser().getUsername());
+          } else {
+            // if connecting to an unsecured portal, no user credentials are needed to authenticate access
+            loadStateTextView.setText("Connected as: Anonymous");
```

Comment:
Connecting anonymously is probably different than not being authenticated at all. If it's a public portal, they'll know it simply because they didn't have to sign in.